esi_leap/common/flocx_market.py

```diff
@@ -0,0 +1,53 @@
+from esi_leap.db import api as dbapi
+from esi_leap.objects import flocx_market_client
+from esi_leap.objects import offer
+from keystoneauth1 import adapter
+from keystoneauth1 import loading as ks_loading
+from oslo_config import cfg
+import datetime
+
+CONF = cfg.CONF
+
+def retrieve_from_flocx_market():
+    auth_plugin = ks_loading.load_auth_from_conf_options(
+        CONF, 'flocx_market')
+    sess = ks_loading.load_session_from_conf_options(CONF, 'flocx_market',
+                                                         auth=auth_plugin)
+    adpt = adapter.Adapter(
+        session=sess,
+        service_type='marketplace',
+        interface='public')
+    marketplace_client = flocx_market_client.FlocxMarketClient(adpt)
+    contract_list = marketplace_client.get_contracts()
+    return contract_list
+
+def update_flocx_market_contract(o_c_r_id):
+    auth_plugin = ks_loading.load_auth_from_conf_options(
+            CONF, 'flocx_market')
+    sess = ks_loading.load_session_from_conf_options(CONF, 'flocx_market',
+                                                         auth=auth_plugin)
+    adpt = adapter.Adapter(
+            session=sess,
+            service_type='marketplace',
+            interface='public')
+    marketplace_client = flocx_market_client.FlocxMarketClient(adpt)
+    res = marketplace_client.update_contract(o_c_r_id)
+
+def split(context, c):
+        # split contract with multiple offers into serval contracts
+    offers = c['offers']
+    o_c_rs = c['o_c_rs']
+    contracts = []
+    for i in range (len(offers)):
+        contract_dict = {}
+        o = offer.Offer.get(context, offers[i]).to_dict()
+        contract_dict['offer_uuid'] = str(o['uuid'])
+        contract_dict['project_id'] = str(o['project_id'])
```

Comment:
This should actually be the bid's project_id. Looks like we need another query...

Reply:
I just pushed the code to have another query to bid.